src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -307,8 +307,13 @@
     @Getter
     private Set<TransitTicket> availableTransitTickets = new HashSet<>();
 
+    /*
+     * Value indicates the number of mapped vehicles for each snapshot. This allows to detect situation
+     * when there is no point in using a snapshot when planning itineraries due to the fact that it was
+     * not available in the historical dataset.
+     */
     @Getter
-    private final Set<SharedVehiclesSnapshotLabel> supportedSnapshotLabels = new HashSet<>();
+    private final Map<SharedVehiclesSnapshotLabel, Long> supportedSnapshotLabels = new HashMap<>();
```

Comment:
Assigning `Long` a value in java is not guaranteed to be atomic - https://stackoverflow.com/questions/4756536/what-operations-in-java-are-considered-atomic/4756578. Please use `Integer` - there won't be that many vehicles anyway.

Reply:
Done